### ISSUE TITLE
Incorrect indentation error

### ISSUE THREAD

**Issue by speekha** [pinterest/ktlint#1089]
When using function with arguments as a delegate to implement an interface, the indentation rule returns an incorrect error.

## Expected Behavior
If I run ktlint check an object looking like this:
```kotlin
object MimeTypeResolver : Map<String, String> by mapOf(
    "application/vnd.hzn-3d-crossword" to ".x3d"
)
```

No error should be returned.

## Observed Behavior
KtLint returns indentation errors:
```
> Task :ktlintCommonMainSourceSetCheck FAILED
File.kt:20:1 Unexpected indentation (4) (should be 8) (indent)
File.kt:21:1 Unexpected indentation (0) (should be 4) (indent)
```

## Steps to Reproduce
Add the previous example to a kotlin project and run a ktlint check.

## Your Environment
* Version of ktlint used: 0.40
* Name and version (or code for custom task) of integration used (Gradle plugin, Maven plugin, command line, custom Gradle task): ktlintCheck (I use the gradle plugin org.jlleitschuh.gradle.ktlint to run the ktlint task in gradle)
* Version of Gradle used (if applicable): 6.8.3
* Operating System and version: Windows (locally), Linux (on CI)
* Link to your project (if it's a public repository): https://github.com/speekha/httpmocker

**Comment by shashachu** [comment]
@speekha thanks for reporting. Can you with with the `--verbose` flag to verify which rule it is that's causing the error?

**Comment by speekha** [comment]
It's the "indent" rule. That's the one I had to disable in my conf to get the check to pass (you can see the .editorconfig conf file on my repo)

**Comment by t-kameyama** [comment]
Try it with version `0.40.0`.
https://github.com/speekha/httpmocker/blob/a39eeaa97fb5e53904828ea5b9ef041264c2312c/gradle/ktlint.gradle#L21

**Comment by speekha** [comment]
@t-kameyama Good catch. I thought I was using 0.40.0 (I've been tinkering with it a bit as I upgraded plugins the other day, I guess I forgot to raise it back to 0.40.0 once I had it working again). 0.40.0 sorted the problem. Sorry for the incorrect report, looks like it's already been fixed.
